src/Mooltipass/MPNode.cpp

```diff
@@ -25,26 +25,35 @@ QByteArray MPNode::EmptyAddress = QByteArray(2, 0);
 
 MPNode::MPNode(const QByteArray &d, QObject *parent, const QByteArray &nodeAddress, const quint32 virt_addr):
     QObject(parent),
-    data(std::move(d)),
-    address(std::move(nodeAddress)),
+    data(d),
+    address(nodeAddress),
```

Comment:
sure about the change? IIRC this is quite a hack to copy the data.....

Reply:
I removed std::move because we were copying it previously too.
For a const lvalue reference (const T&) std::move is only casting to const T&&, which is still calling the copy constructor.
https://stackoverflow.com/questions/38917130/does-stdmove-work-with-lvalue-references-how-does-stdmove-work-on-standard

Reply:
I pushed a solution for this to move QByteArray whenever possible.
There are 3 scenario to create MPNode in MPDevice:
- **const T& case**: `MPNode *pnode = pMesProt->createMPNode(this, address);`
    - Here `address` is given from function parameter (const T&) and used later.
    - After moving here would still copy the QByteArray
- **Temporary QByteArray**: `temp_node = pMesProt->createMPNode(temp_node_pt->getNodeData(), this, temp_node_pt->getAddress(), temp_node_pt->getVirtualAddress());`
   - Because of the RVO the new function with universal reference (T&&) will be called by default for MPNode and it will move the QByteArray
- **QByteArray as lvalue and not used later**: `MPNode* importedNode = pMesProt->createMPNode(dataCore, this, serviceAddr, 0);`
   - If I move the lvalue the correct function is called and it is moving the QByteArray
   - Changed to: `MPNode* importedNode = pMesProt->createMPNode(qMove(dataCore), this, qMove(serviceAddr), 0);`

Reply:
calling @raoulh  on this one

Reply:
@deXol is right